model/Check/System/AdvancedSearch/AdvancedSearchIndexesCheck.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoSystemStatus\model\Check\System\AdvancedSearch;
+
+use Throwable;
+use common_report_Report;
+use oat\tao\helpers\Template;
+use oat\oatbox\reporting\Report;
+use oat\tao\model\search\SearchProxy;
+use oat\tao\elasticsearch\ElasticSearch;
+use oat\tao\model\AdvancedSearch\AdvancedSearchChecker;
+use oat\taoAdvancedSearch\model\Index\Report\IndexSummarizer;
+
+class AdvancedSearchIndexesCheck extends AbstractAdvancedSearchCheck
+{
+    public const PARAM_ALLOWABLE_PERCENTAGE = 'allowablePercentage';
+    public const PARAM_BLACKLISTED_INDEXES = 'blacklistedIndexes';
+
+    /** @var float */
+    private $allowablePercentage;
+
+    public function getDetails(): string
+    {
+        return __('Indexes population');
+    }
+
+    protected function doCheck(): common_report_Report
+    {
+        if (!$this->getAdvancedSearchChecker()->isEnabled()) {
+            return Report::createError('Advanced search disabled');
+        }
+
+        $advancedSearch = $this->getSearchProxy()->getAdvancedSearch();
+
+        if (!$advancedSearch instanceof ElasticSearch || !$advancedSearch->ping()) {
+            return Report::createError('Advanced search unavailable');
+        }
+
+        try {
+            $reports = [];
+            $summary = $this->getIndexSummarizer()->summarize();
```

Comment:
- Did we check if the queries from summarize wont harm the server? This will be executed every 5min.